shared/clusterconfig.go

```diff
@@ -84,23 +89,16 @@ type Pod struct {
 	ReadinessGates string
 }
 
-// setConfig gets env configs and sets on local vars.
-func setConfig() {
-	product = os.Getenv("ENV_PRODUCT")
-	module = os.Getenv("ENV_MODULE")
-}
-
 // ClusterConfig returns a singleton cluster with all terraform config and vars.
-func ClusterConfig() *Cluster {
-	setConfig()
+func ClusterConfig(cfg *config.Product) *Cluster {
```

Comment:
We should think about renaming these vars `config.Product` and have a better way to structure these. Its confusing to me when I see this var `cfg *config.Product` and then `cfg.Product`/`cfg.Module`... these are env config so we should rename them as such like `envConf *config.Env`, followed by `envConf.Product`/`envConf.Module`